gql/builder.go

```diff
@@ -25,6 +24,9 @@ const (
 	// QueryReporterContextKey is the key used with context.WithValue to locate the QueryReporter.
 	QueryReporterContextKey = "GraphQLQueryReporter"
 
+	// QueryFunctionReporterContextKey is the key used with context.WithValue to locate the QueryReporter.
+	QueryFunctionReporterContextKey = "GraphQLQueryFunctionReporter"
```

Comment:
Sorry for not being clear but I don't think you necessarily need a new key. That would be painful because then you will need to put two structs into the context for this to work.